org.eclipse.lsp4e/src/org/eclipse/lsp4e/LSPEclipseUtils.java

```diff
@@ -562,27 +563,46 @@ public static IDocument getDocument(@Nullable IResource resource) {
 		if (resource == null) {
 			return null;
 		}
+		IDocument document = null;
+		synchronized(connectedFiles) {
+			document = getExistingDocument(resource);
 
-		IDocument document = getExistingDocument(resource);
+			if (document == null && resource.getType() == IResource.FILE) {
+				ITextFileBufferManager bufferManager = FileBuffers.getTextFileBufferManager();
+				if (bufferManager == null)
```

Comment:
to be consistent with the rest of the codebase, could you add `{}` around the `return document` statement?